glotpress.php

```diff
@@ -164,12 +164,28 @@ function gp_unsupported_permalink_structure_admin_notice() {
 
 require_once GP_PATH . 'gp-settings.php';
 
+/*
+ * Check if we need to run the upgrade routine, but only run it on the admin side.
+ */
+if ( is_admin() && GP_DB_VERSION > get_option( 'gp_db_version' ) ) {
+	require_once GP_PATH . GP_INC . 'install-upgrade.php';
+	gp_upgrade_db();
+}
+
 /**
  * Perform necessary actions on activation.
  *
  * @since 1.0.0
  */
 function gp_activate_plugin() {
+	// Create/upgrade database tables.
+	// Even though there is logic above to upgrade the database when a user visits an admin page, the plugin should be
+	// functional regardless of whether the user visited an admin page or not.
+	// This also makes it possible to activate the plugin through wp-cli, since in that case, the database upgrade logic
+	// above will not run since is_admin() is false.
+	require_once GP_PATH . GP_INC . 'install-upgrade.php';
+	gp_upgrade_db();
```

Comment:
If the user activates and deactivates the plugin more than once, the second and subsequent times it is not necessary to run this process, only if the database version has been updated.

```suggestion
	if ( GP_DB_VERSION > get_option( 'gp_db_version' ) {
		require_once GP_PATH . GP_INC . 'install-upgrade.php';
		gp_upgrade_db();
	}
```

Reply:
Good point, I pushed some commits that fix this.